File: application/src/main/java/com/stayrascal/service/application/event/EventService.java
package com.stayrascal.service.application.event;

import com.stayrascal.service.application.domain.Event;

import java.util.List;

public interface EventService {
  Event addEvent(Event event);

  List<Event> listEvents();

  List<Event> getEventsByUserId(long userId);
}

File: application/src/main/scala/com/stayrascal/service/application/events/EventService.scala
package com.stayrascal.service.application.events

import org.apache.spark.streaming.dstream.DStream

trait EventStreamService {
  def init(): Unit

  def addEvent(event: String)

  def getEventStream: DStream[(Long, Long, String, String)]

  def saveEvent(eventStream: DStream[(Long, Long, String, String)])
}
